$MirOS: ports/mailnews/pine/patches/patch-imap_src_osdep_unix_env_unix_c,v 1.1.7.1 2005/03/18 15:49:48 tg Exp $
$OpenBSD: patch-imap_src_osdep_unix_env_unix_c,v 1.3 2003/10/05 21:42:29 jakob Exp $
--- imap/src/osdep/unix/env_unix.c.orig	Mon Sep 13 21:30:57 2004
+++ imap/src/osdep/unix/env_unix.c	Wed May  4 23:28:08 2005
@@ -382,7 +382,7 @@ static void do_date (char *date,char *pr
     date += strlen (date);	/* make next sprintf append */
   }
 				/* output the date */
-  sprintf (date,fmt,t->tm_mday,months[t->tm_mon],t->tm_year+1900,
+  sprintf (date,fmt,t->tm_mday,months[t->tm_mon],(long)t->tm_year+1900,
 	   t->tm_hour,t->tm_min,t->tm_sec,zone/60,abs (zone) % 60);
 				/* append timezone suffix if desired */
   if (suffix) rfc822_timezone (date,(void *) t);
@@ -394,7 +394,7 @@ static void do_date (char *date,char *pr
 
 void rfc822_date (char *date)
 {
-  do_date (date,"%s, ","%d %s %d %02d:%02d:%02d %+03d%02d",
+  do_date (date,"%s, ","%d %s %ld %02d:%02d:%02d %+03d%02d",
 	   no822tztext ? NIL : T);
 }
 
@@ -405,7 +405,7 @@ void rfc822_date (char *date)
 
 void rfc822_fixed_date (char *date)
 {
-  do_date (date,NIL,"%02d %s %4d %02d:%02d:%02d %+03d%02d",NIL);
+  do_date (date,NIL,"%02d %s %4ld %02d:%02d:%02d %+03d%02d",NIL);
 }
 
 
@@ -415,7 +415,7 @@ void rfc822_fixed_date (char *date)
 
 void internal_date (char *date)
 {
-  do_date (date,NIL,"%02d-%s-%d %02d:%02d:%02d %+03d%02d",NIL);
+  do_date (date,NIL,"%02d-%s-%ld %02d:%02d:%02d %+03d%02d",NIL);
 }
 
 /* Initialize server
@@ -532,7 +532,23 @@ static struct passwd *valpwd (char *user
       fs_give ((void **) &s);
     }
   }
+#ifdef BSO_AUTH
+  else if (s = strchr(user, ':')) {
+    *s++ = '\0';
+    if (pw = pwuser (user)) {
+      pw->pw_class = s;
+      s = cpystr (pw->pw_name);	/* copy returned name in case we need it */
+      if (*pwd && !(ret = checkpw (pw,pwd,argc,argv)) &&
+	  (*pwd == ' ') && pwd[1] && (ret = pwuser (s)))
+        ret = checkpw (pw,pwd+1,argc,argv);
+      fs_give ((void **) &s);	/* don't need copy of name any more */
+    }
+  }
+#endif
   else if (pw = pwuser (user)) {/* can get user? */
+#ifdef BSO_AUTH
+    pw->pw_class = NIL;
+#endif
     s = cpystr (pw->pw_name);	/* copy returned name in case we need it */
     if (*pwd && !(ret = checkpw (pw,pwd,argc,argv)) &&
 	(*pwd == ' ') && pwd[1] && (ret = pwuser (s)))
@@ -809,15 +825,12 @@ char *myhomedir ()
 static char *mymailboxdir ()
 {
   char *home = myhomedir ();
-				/* initialize if first time */
-  if (!myMailboxDir && myHomeDir) {
     if (mailsubdir) {
       char tmp[MAILTMPLEN];
-      sprintf (tmp,"%s/%s",home,mailsubdir);
+      snprintf (tmp,MAILTMPLEN,"%s/%s",home,mailsubdir);
       myMailboxDir = cpystr (tmp);/* use pre-defined subdirectory of home */
     }
     else myMailboxDir = cpystr (home);
-  }
   return myMailboxDir ? myMailboxDir : "";
 }
 
